mysqli.php

```diff
@@ -423,7 +423,7 @@ function sql_freeresult($query_id = false)
 		{
 			$query_id = $this->query_result;
 		}
-		if ($query_id) {
+		if ($query_id && isset($query_id->num_rows)) {
```

Comment:
I'm not sure why you've changed this line. Doesn't this has the possibility to leave some ressources used instead of being freed?

Reply:
OK, I got it after testing: otherwise we end up trying to free already closed queries, and have an error.